packages/rich-text/src/plugins/EmbeddedEntityBlock/Util.ts

```diff
@@ -21,19 +22,22 @@ export async function selectEntityAndInsert(
     baseConfig.entityType === 'Asset' ? dialogs.selectSingleAsset : dialogs.selectSingleEntry;
   const config = { ...baseConfig, withCreate: true };
 
-  const { selection } = editor;
+  const rteSlide = watchCurrentSlide(sdk.navigator);
   const entity = await selectEntity(config);
+
   if (!entity) {
     logAction('cancelCreateEmbedDialog', { nodeType });
-    return;
+  } else {
+    insertBlock(editor, nodeType, entity);
+    ensureFollowingParagraph(editor);
+    logAction('insert', { nodeType });
   }
-
-  select(editor, selection);
```

Comment:
@aodhagan-cf @MayaGillilanContentful @YvesRijckaert any thoughts on removing this? I did all kind of tests, but couldn't see any negative impact. Same goes for Cypress tests which are still running.

This line seems odd to me, maybe it did serve a purpose in a past Slate/Plate version, but right now I don't see how the selection should be gone and why we have to reset it, it seems like ne only thing that's necessary is to re-focus the editor and the cursor position is in the right place.

Reply:
I would say if it passes unit tests and you can't see anything glaringly obvious on problems it causes happy for you to remove it. If it is a problem in future, we write a test to capture that

Reply:
Yes, if nothing pops up, we can probably remove it in a whole lot of places. But waiting for @z0al perhaps he can shed some light.